### ISSUE TITLE
pyamqp write timeout when writing to a different Azure Datacenter

### ISSUE THREAD

**Issue by eeertel** [Azure/azure-sdk-for-python#30425]
- **Package Name**:azure-servicebus 
- **Package Version**: 7.10.0
- **Operating System**:20~20.04.1-Ubuntu SMP Fri Aug 5 12:16:53 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
- **Python Version**: Python 3.8.10

**Describe the bug**
When sending a large message (250k - see attached code snippet) from the JP-West Azure Datacenter to EU-West Azure Datacenter, the send operation times out

**To Reproduce**
1. Execute the code snipped below:
```
from azure.servicebus import ServiceBusClient,ServiceBusMessage,_servicebus_sender
import logging
import os
import sys
logger = logging.getLogger('azure.servicebus')
logger.setLevel(logging.DEBUG)
handler = logging.StreamHandler(stream=sys.stdout)
logger.addHandler(handler)

queue_name = os.environ['SB_QUEUE_NAME']
conn_str = os.environ['SB_CONN_STR']

service_bus_client =  ServiceBusClient.from_connection_string(conn_str = conn_str, logging_enable = False)
servicebus_sender = service_bus_client.get_queue_sender(queue_name)

with service_bus_client:
    payload = "A" * 250 * 1024
    servicebus_sender.send_messages(ServiceBusMessage(payload))
```

**Expected behavior**
The send operation works without any problems with the previous version (7.9.0)

**Screenshots**
Script output (last lines only):
```
Connection state changed: <ConnectionState.OPEN_SENT: 7> -> <ConnectionState.OPENED: 9>
Session state changed: <SessionState.BEGIN_SENT: 1> -> <SessionState.MAPPED: 3>
Link state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
Management link receiver state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
Link state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
Management link sender state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
CBS completed opening with status: <ManagementOpenResult.OK: 1>
CBS update in progress. Token put time: 1684321763
CBS update in progress. Token put time: 1684321763
CBS Put token result (<ManagementExecuteOperationResult.OK: 0>), status code: 202, status_description: b'Accepted'.
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
Link state changed: <LinkState.DETACHED: 0> -> <LinkState.ATTACH_SENT: 1>
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
Link state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
An error occurred when detaching the link: AMQPConnectionError('Error condition: ErrorCondition.SocketError\n Error Description: Can not send frame out due to exception: The write operation timed out')
Link state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACHED: 0>
An error occurred when detaching the link: AMQPConnectionError('Error condition: ErrorCondition.SocketError\n Error Description: Can not send frame out due to exception: The write operation timed out')
Link state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACHED: 0>
Management link receiver state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACHED: 0>
An error occurred when detaching the link: AMQPConnectionError('Error condition: ErrorCondition.SocketError\n Error Description: Can not send frame out due to exception: The write operation timed out')
Link state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACHED: 0>
Management link sender state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACHED: 0>
An error occurred when ending the session: AMQPConnectionError('Error condition: ErrorCondition.SocketError\n Error Description: Can not send frame out due to exception: The write operation timed out')
Session state changed: <SessionState.MAPPED: 3> -> <SessionState.UNMAPPED: 0>
An error occurred when closing the connection: AMQPConnectionError('Error condition: ErrorCondition.SocketError\n Error Description: Can not send frame out due to exception: The write operation timed out')
Connection state changed: <ConnectionState.OPENED: 9> -> <ConnectionState.END: 13>
Connection state changed: None -> <ConnectionState.START: 0>
Connection state changed: <ConnectionState.START: 0> -> <ConnectionState.HDR_SENT: 2>
Connection state changed: <ConnectionState.HDR_SENT: 2> -> <ConnectionState.HDR_SENT: 2>
Connection state changed: <ConnectionState.HDR_SENT: 2> -> <ConnectionState.OPEN_PIPE: 4>
Session state changed: <SessionState.UNMAPPED: 0> -> <SessionState.BEGIN_SENT: 1>
Link state changed: <LinkState.DETACHED: 0> -> <LinkState.ATTACH_SENT: 1>
Management link receiver state changed: <LinkState.DETACHED: 0> -> <LinkState.ATTACH_SENT: 1>
Link state changed: <LinkState.DETACHED: 0> -> <LinkState.ATTACH_SENT: 1>
Management link sender state changed: <LinkState.DETACHED: 0> -> <LinkState.ATTACH_SENT: 1>
Connection state changed: <ConnectionState.OPEN_PIPE: 4> -> <ConnectionState.OPEN_SENT: 7>
Connection state changed: <ConnectionState.OPEN_SENT: 7> -> <ConnectionState.OPENED: 9>
Session state changed: <SessionState.BEGIN_SENT: 1> -> <SessionState.MAPPED: 3>
Link state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
Management link receiver state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
Link state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
Management link sender state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
CBS completed opening with status: <ManagementOpenResult.OK: 1>
CBS update in progress. Token put time: 1684321769
CBS update in progress. Token put time: 1684321769
CBS Put token result (<ManagementExecuteOperationResult.OK: 0>), status code: 202, status_description: b'Accepted'.
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
Link state changed: <LinkState.DETACHED: 0> -> <LinkState.ATTACH_SENT: 1>
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
Link state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
AMQP error occurred: (AMQPConnectionError('Error condition: ErrorCondition.SocketError\n Error Description: Can not send frame out due to exception: The write operation timed out')), condition: (<ErrorCondition.SocketError: b'amqp:socket-error'>), description: ('Can not send frame out due to exception: The write operation timed out').
An error occurred when detaching the link: AMQPConnectionError('Error condition: ErrorCondition.SocketError\n Error Description: Can not send frame out due to exception: The write operation timed out')
Link state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACHED: 0>
An error occurred when detaching the link: AMQPConnectionError('Error condition: ErrorCondition.SocketError\n Error Description: Can not send frame out due to exception: The write operation timed out')
Link state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACHED: 0>
Management link receiver state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACHED: 0>
An error occurred when detaching the link: AMQPConnectionError('Error condition: ErrorCondition.SocketError\n Error Description: Can not send frame out due to exception: The write operation timed out')
Link state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACHED: 0>
Management link sender state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACHED: 0>
An error occurred when ending the session: AMQPConnectionError('Error condition: ErrorCondition.SocketError\n Error Description: Can not send frame out due to exception: The write operation timed out')
Session state changed: <SessionState.MAPPED: 3> -> <SessionState.UNMAPPED: 0>
An error occurred when closing the connection: AMQPConnectionError('Error condition: ErrorCondition.SocketError\n Error Description: Can not send frame out due to exception: The write operation timed out')
Connection state changed: <ConnectionState.OPENED: 9> -> <ConnectionState.END: 13>
'servicebus.pysdk-c5534f85' operation has exhausted retry. Last exception: ServiceBusConnectionError('Can not send frame out due to exception: The write operation timed out Error condition: amqp:socket-error.').
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 835, in close
    self._outgoing_close(error=error)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 442, in _outgoing_close
    self._send_frame(0, close_frame)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 265, in _send_frame
    raise self._error
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/session.py", line 462, in end
    self._outgoing_end(error=error)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/session.py", line 167, in _outgoing_end
    self._connection._process_outgoing_frame(  # pylint: disable=protected-access
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 651, in _process_outgoing_frame
    self._send_frame(channel, frame)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 265, in _send_frame
    raise self._error
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/link.py", line 251, in detach
    self._outgoing_detach(close=close, error=error)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/link.py", line 213, in _outgoing_detach
    self._session._outgoing_detach(detach_frame) # pylint: disable=protected-access
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/session.py", line 410, in _outgoing_detach
    self._connection._process_outgoing_frame(  # pylint: disable=protected-access
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 651, in _process_outgoing_frame
    self._send_frame(channel, frame)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 265, in _send_frame
    raise self._error
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/link.py", line 251, in detach
    self._outgoing_detach(close=close, error=error)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/link.py", line 213, in _outgoing_detach
    self._session._outgoing_detach(detach_frame) # pylint: disable=protected-access
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/session.py", line 410, in _outgoing_detach
    self._connection._process_outgoing_frame(  # pylint: disable=protected-access
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 651, in _process_outgoing_frame
    self._send_frame(channel, frame)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 265, in _send_frame
    raise self._error
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/link.py", line 251, in detach
    self._outgoing_detach(close=close, error=error)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/link.py", line 213, in _outgoing_detach
    self._session._outgoing_detach(detach_frame) # pylint: disable=protected-access
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/session.py", line 410, in _outgoing_detach
    self._connection._process_outgoing_frame(  # pylint: disable=protected-access
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 651, in _process_outgoing_frame
    self._send_frame(channel, frame)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 265, in _send_frame
    raise self._error
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_base_handler.py", line 411, in _do_retryable_operation
    return operation(**kwargs)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_servicebus_sender.py", line 276, in _send
    self._amqp_transport.send_messages(self, message, _LOGGER, timeout=timeout, last_exception=last_exception)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_transport/_pyamqp_transport.py", line 487, in send_messages
    sender._handler.send_message(message._message, timeout=timeout) # pylint:disable=protected-access
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/client.py", line 685, in send_message
    self._do_retryable_operation(self._send_message_impl, message=message, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/client.py", line 284, in _do_retryable_operation
    raise retry_settings["history"][-1]
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/client.py", line 263, in _do_retryable_operation
    return operation(*args, timeout=absolute_timeout, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/client.py", line 658, in _send_message_impl
    running = self.do_work()
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/client.py", line 413, in do_work
    return self._client_run(**kwargs)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/client.py", line 588, in _client_run
    self._link.update_pending_deliveries()
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/sender.py", line 157, in update_pending_deliveries
    sent_and_settled = self._outgoing_transfer(delivery)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/sender.py", line 107, in _outgoing_transfer
    self._session._outgoing_transfer(  # pylint:disable=protected-access
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/session.py", line 326, in _outgoing_transfer
    self._connection._process_outgoing_frame(  # pylint: disable=protected-access
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 651, in _process_outgoing_frame
    self._send_frame(channel, frame)
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_pyamqp/_connection.py", line 265, in _send_frame
    raise self._error
azure.servicebus._pyamqp.error.AMQPConnectionError: Error condition: ErrorCondition.SocketError
 Error Description: Can not send frame out due to exception: The write operation timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/teamplay/bin/testsbupload.py", line 18, in <module>
    servicebus_sender.send_messages(ServiceBusMessage(payload))
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_servicebus_sender.py", line 482, in send_messages
    self._do_retryable_operation(
  File "/usr/local/lib/python3.8/dist-packages/azure/servicebus/_base_handler.py", line 428, in _do_retryable_operation
    raise last_exception
azure.servicebus.exceptions.ServiceBusConnectionError: Can not send frame out due to exception: The write operation timed out Error condition: amqp:socket-error.

```

**Additional context**
The same code works well sending the same payload to a servicebus located in the Azure China East2 datacenter

```
Connection state changed: None -> <ConnectionState.START: 0>
Connection state changed: <ConnectionState.START: 0> -> <ConnectionState.HDR_SENT: 2>
Connection state changed: <ConnectionState.HDR_SENT: 2> -> <ConnectionState.HDR_SENT: 2>
Connection state changed: <ConnectionState.HDR_SENT: 2> -> <ConnectionState.OPEN_PIPE: 4>
Session state changed: <SessionState.UNMAPPED: 0> -> <SessionState.BEGIN_SENT: 1>
Link state changed: <LinkState.DETACHED: 0> -> <LinkState.ATTACH_SENT: 1>
Management link receiver state changed: <LinkState.DETACHED: 0> -> <LinkState.ATTACH_SENT: 1>
Link state changed: <LinkState.DETACHED: 0> -> <LinkState.ATTACH_SENT: 1>
Management link sender state changed: <LinkState.DETACHED: 0> -> <LinkState.ATTACH_SENT: 1>
Connection state changed: <ConnectionState.OPEN_PIPE: 4> -> <ConnectionState.OPEN_SENT: 7>
Connection state changed: <ConnectionState.OPEN_SENT: 7> -> <ConnectionState.OPENED: 9>
Session state changed: <SessionState.BEGIN_SENT: 1> -> <SessionState.MAPPED: 3>
Link state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
Management link receiver state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
Link state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
Management link sender state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
CBS completed opening with status: <ManagementOpenResult.OK: 1>
CBS update in progress. Token put time: 1684322478
CBS update in progress. Token put time: 1684322478
CBS Put token result (<ManagementExecuteOperationResult.OK: 0>), status code: 202, status_description: b'Accepted'.
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
Link state changed: <LinkState.DETACHED: 0> -> <LinkState.ATTACH_SENT: 1>
Link state changed: <LinkState.ATTACH_SENT: 1> -> <LinkState.ATTACHED: 3>
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
CBS status check: state == <CbsAuthState.OK: 0>, expired == False, refresh required == False
Link state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACH_SENT: 4>
Link state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACH_SENT: 4>
Management link receiver state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACH_SENT: 4>
Link state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACH_SENT: 4>
Management link sender state changed: <LinkState.ATTACHED: 3> -> <LinkState.DETACH_SENT: 4>
An error occurred when detaching the link: AMQPConnectionError('Error condition: ErrorCondition.InternalError\n Error Description: Link already closed.')
Link state changed: <LinkState.DETACH_SENT: 4> -> <LinkState.DETACHED: 0>
Management link sender state changed: <LinkState.DETACH_SENT: 4> -> <LinkState.DETACHED: 0>
An error occurred when detaching the link: AMQPConnectionError('Error condition: ErrorCondition.InternalError\n Error Description: Link already closed.')
Link state changed: <LinkState.DETACH_SENT: 4> -> <LinkState.DETACHED: 0>
Management link receiver state changed: <LinkState.DETACH_SENT: 4> -> <LinkState.DETACHED: 0>
An error occurred when detaching the link: AMQPConnectionError('Error condition: ErrorCondition.InternalError\n Error Description: Link already closed.')
Link state changed: <LinkState.DETACH_SENT: 4> -> <LinkState.DETACHED: 0>
Session state changed: <SessionState.MAPPED: 3> -> <SessionState.END_SENT: 4>
Connection state changed: <ConnectionState.OPENED: 9> -> <ConnectionState.CLOSE_SENT: 11>
Connection state changed: <ConnectionState.CLOSE_SENT: 11> -> <ConnectionState.END: 13>
Session state changed: <SessionState.END_SENT: 4> -> <SessionState.DISCARDING: 6>
Link state changed: <LinkState.DETACHED: 0> -> <LinkState.DETACHED: 0>
Management link sender state changed: <LinkState.DETACHED: 0> -> <LinkState.DETACHED: 0>
Link state changed: <LinkState.DETACHED: 0> -> <LinkState.DETACHED: 0>
Management link receiver state changed: <LinkState.DETACHED: 0> -> <LinkState.DETACHED: 0>
Link state changed: <LinkState.DETACHED: 0> -> <LinkState.DETACHED: 0>
```

**Comment by kashifkhan** [comment]
Thank you for the feedback @eeertel . 

As I try to reproduce this, can you try passing in a timeout of say 5 seconds or more in `send_messages` to see if that addresses things in meantime. 

As a follow up, the case where the message is sent properly is going from JP-West to China-East?

**Comment by eeertel** [comment]
@kashifkhan 
yes - sending from JP-West to China-East works

Have tried setting the timeout to 5 (and even 60) - without any changes in the behavior (still failing)
Seems that everything with a payload below 64k works well between JP-West and EU-West. The error appears with packet sizes >= 64k

**Comment by kashifkhan** [comment]
Thanks for that @eeertel, Im assuming this is service bus premium ?

**Comment by eeertel** [comment]
@kashifkhan - both tiers (China-East and EU-West) are **Standard** pricing tier

**Comment by l0lawrence** [comment]
Hey @eeertel would you be able to email logs with logging_enable set to true on the client?

`service_bus_client =  ServiceBusClient.from_connection_string(conn_str = conn_str, logging_enable = True)`

It would be to @kashifkhan's username at microsoft.com

**Comment by eeertel** [comment]
mail has been sent - please let me know if you need any additional details or want to have a short Teams meetings

**Comment by eeertel** [comment]
@l0lawrence :
purged the queue and tried a re-send. Unfortunately no message appears in the queue (checked both in portal and in the standalone ServiceBusExplorer)

When I send a "short" message (length < 64k), no error is thrown and the message is visible in the queue

**Comment by l0lawrence** [comment]
Hey @eeertel, we were able to repro and are working on a fix. We will link the PR as soon as it is available.

**Comment by eeertel** [comment]
@kashifkhan : Thank you - the 

> Hi @eeertel, we have released an update of service bus, version 7.11.0, which has a fix for this. We expose a `socket_timeout` kwarg that will let you change the timeout value for the underlying socket when reading and writing.

Thank you ! I have tested the new version for the last two days and I can confirm that it solves all the known problems.
